src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -181,6 +181,7 @@ public function setValue(&$objectOrArray, $propertyPath, $value)
                     if ($propertyPath->isIndex($i)) {
                         if ($overwrite = !isset($zval[self::REF])) {
                             $ref = &$zval[self::REF];
+                            $ref = $zval[self::VALUE];
```

Comment:
Not a typo: the prev line turns `$zval[self::REF]` into a reference, which required for writeIndex to work.